src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,157 @@
+import org.broadinstitute.dsm.db.dao.kit.BSPKitQueryResultDao;
+import org.broadinstitute.dsm.db.dto.kit.BSPKitQueryResultDto;
+import org.broadinstitute.dsm.db.dto.settings.InstanceSettingsDto;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitStatus;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public Optional<BSPKitStatus> getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        Optional<BSPKitStatus> result = Optional.empty();
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId())) {
+                String message = "Kit of exited participant " + maybeBspKitQueryResult.getBspParticipantId() + " was received by GP.<br>";
+                notificationUtil.sentNotification(maybeBspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.DSM_SUBJECT);
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.EXITED));
+            }
+            else if (StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.DEACTIVATED));
+            }
+        }
+        return result;
+    }
+
+    public boolean canReceiveKit(@NonNull String kitLabel) {
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return false;
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId()) || StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                logger.info("Kit can not be received.");
+                return false;
+            }
+        }
+        return true;
+
+    }
+
+    public Optional<BSPKitInfo> receiveBSPKit(String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Trying to receive kit " + kitLabel);
+        return TransactionWrapper.inTransaction(conn -> {
```

Comment:
try to cut direct connections as much as possible, anyway such technique should be eliminated in whole project and if we do it slowly, step by step, will make refactoring process much easier.